src/modules/adaptors/Staking/StakingAdaptor.sol

```diff
@@ -0,0 +1,383 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { ERC20, SafeTransferLib, Cellar, PriceRouter, Registry, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import { IWETH9 } from "src/interfaces/external/IWETH9.sol";
+import { BaseAdaptor, ERC20, SafeTransferLib } from "src/modules/adaptors/BaseAdaptor.sol";
+
+/**
+ * @title Staking Adaptor
+ * @notice Serves as a universal template for a variety of staking adaptors.
+ * @dev Allows inheriting adaptors to implement staking, unstaking, wrapping, unwrapping.
```

Comment:
Add a note how ERC20 Positions keep track of `staked` tokens, whereas withdrawals (via queue NFT requests) are used to account for tokens in withdrawal state.

Reply:
Done!